app/views/govuk_publishing_components/components/_intervention.html.erb

```diff
@@ -35,14 +40,16 @@
     suggestion_link_text = intervention_helper.accessible_text
   end
 
+  data_attributes[:module] = "intervention"
+
   section_options = {
     class: "gem-c-intervention",
     role: "region", aria: aria_attributes,
     data: data_attributes,
   }
   section_options.merge!({ hidden: true }) if hide
 %>
-<% if suggestion_text || (suggestion_link_text && suggestion_link_url) %>
+<% if intervention_helper.show? %>
```

Comment:
There's a lot of logic before this thing that decides whether the component gets shown or not - might be better if this line was earlier. Maybe outside the scope of this PR.

Reply:
I can only get it to this point, but it's less readable:

```
<%
  dismiss_text ||= false
  suggestion_link_text ||= false
  suggestion_link_url ||= false
  suggestion_text ||= nil
  hide ||= false
  new_tab ||= false

  data_attributes ||= {}
  suggestion_data_attributes ||= {}
  dismiss_data_attributes ||= {}
  dismiss_data_attributes

  aria_attributes ||= {}
  aria_attributes[:label] = 'Intervention'

  options = {
    params: request.params,
    query_string: request.query_string,
    suggestion_text: suggestion_text,
    suggestion_link_text: suggestion_link_text,
    suggestion_link_url: suggestion_link_url,
  }

  intervention_helper = GovukPublishingComponents::Presenters::InterventionHelper.new(options)
  return unless intervention_helper.show?

  dismiss_href = intervention_helper.dismiss_link

  suggestion_tag_options = {
    class: "govuk-link",
    href: suggestion_link_url,
    data: suggestion_data_attributes,
  }

  data_attributes[:module] = "intervention"

  section_options = {
    class: "gem-c-intervention",
    role: "region", aria: aria_attributes,
    data: data_attributes,
  }
  section_options.merge!({ hidden: true }) if hide

  if new_tab && (suggestion_link_text && suggestion_link_url)
    target = "_blank"
    rel = intervention_helper.security_attr

    suggestion_tag_options.merge!({ target: target, rel: rel })

    suggestion_link_text = intervention_helper.accessible_text
  end
%>

<%= tag.section **section_options do %>
  <p class="govuk-body">
    <%= tag.span suggestion_text, class: "gem-c-intervention__textwrapper" if suggestion_text %>
    <% if suggestion_link_text && suggestion_link_url %>
      <%= tag.a suggestion_link_text, **suggestion_tag_options %>
    <% end %>
  </p>

  <% if dismiss_text %>
    <p class="govuk-body">
      <%= tag.a class: "govuk-link gem-c-intervention__dismiss-link", href: dismiss_href, data: dismiss_data_attributes do %>
        <svg class="gem-c-intervention__dismiss-icon"
          width="19" height="19" viewBox="0 0 19 19"
          aria-hidden="true"
          fill="none" stroke="currentColor">
          <path d="M0.499997 9.5C0.499997 4.52944 4.52944 0.499999 9.5 0.499999C14.4706 0.5 18.5 4.52944 18.5 9.5C18.5 14.4706 14.4706 18.5 9.5 18.5C4.52943 18.5 0.499997 14.4706 0.499997 9.5Z"/>
          <path d="M13.3477 5.4209L9.67383 9.09473L6 5.4209"/>
          <path d="M6 13.4209L9.67383 9.74707L13.3477 13.4209"/>
        </svg>
        <%= dismiss_text %>
      <% end %>
    </p>
  <% end %>
<% end %>
```